<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Build OmniPathR databases â€¢ strpip</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Build OmniPathR databases">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">strpip</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/strpip.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/database.html">Build OmniPathR databases</a></li>
    <li><a class="dropdown-item" href="../articles/interoperability.html">Interoperability with file formats</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/hungms/strpip/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Build OmniPathR databases</h1>
                        <h4 data-toc-skip class="author">Author: Matthew
Hung</h4>
            
            <h4 data-toc-skip class="date">Last Updated: 2025/04/02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hungms/strpip/blob/main/vignettes/database.Rmd" class="external-link"><code>vignettes/database.Rmd</code></a></small>
      <div class="d-none name"><code>database.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">600</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hungms.github.io/deseq2pip/" class="external-link">strpip</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'magrittr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:rlang':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     set_names</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:tidyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     extract</span></span></code></pre>
<pre><code><span><span class="co">## strpip 0.1.0</span></span></code></pre>
<div class="section level2">
<h2 id="retrieve-biomrt-database">Retrieve biomRt database<a class="anchor" aria-label="anchor" href="#retrieve-biomrt-database"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import biomart database - this returns the biomart_dict directly now</span></span>
<span><span class="va">biomart_dict</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_biomart_local.html">import_biomart_local</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># biomart_dict is a data frame that contains the biomart mouse and human gene symbols, and their corresponding chromosome</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">biomart_dict</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   mouse_gene_symbol mouse_chromosome human_gene_symbol human_chromosome</span></span>
<span><span class="co">## 1            mt-Co3               MT            MT-CO3               MT</span></span>
<span><span class="co">## 2           mt-Atp6               MT           MT-ATP6               MT</span></span>
<span><span class="co">## 3            mt-Nd3               MT            MT-ND3               MT</span></span>
<span><span class="co">## 4            mt-Nd4               MT            MT-ND4               MT</span></span>
<span><span class="co">## 5            mt-Nd1               MT            MT-ND1               MT</span></span>
<span><span class="co">## 6            mt-Nd2               MT            MT-ND2               MT</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># save biomart_dict to TSV file - only run this if you need to update the data</span></span>
<span><span class="co"># write.table(biomart_dict, "inst/extdata/release-105.tsv", col.names = TRUE, row.names = FALSE, quote = FALSE, sep = "\t")</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-omnipathr-databases-for-gene-annotation">Build OmniPathR databases for gene annotation<a class="anchor" aria-label="anchor" href="#build-omnipathr-databases-for-gene-annotation"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Only run the following code if you need to update the OmniPath databases</span></span>
<span><span class="co"># Make sure OmniPathR is installed with: BiocManager::install("OmnipathR")</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"OmnipathR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"This vignette requires the OmnipathR package. Please install it with BiocManager::install('OmnipathR')"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># import OmniPathR databases for human gene annotation</span></span>
<span><span class="va">human.omnipath.db</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/intercell.html" class="external-link">import_omnipath_intercell</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">human.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add human_gene_symbol and database_cs columns</span></span>
<span><span class="va">human.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">human.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>        human_gene_symbol <span class="op">=</span>  <span class="va">genesymbol</span>,</span>
<span>        database_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">database</span>, <span class="st">"UniProt"</span><span class="op">)</span>, <span class="st">"UniProt"</span>, <span class="va">database</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">biomart_dict</span>, <span class="va">.</span>, by <span class="op">=</span> <span class="st">"human_gene_symbol"</span>, all <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genesymbol"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select columns of interest for gene annotation</span></span>
<span><span class="co"># annotate genes that are transmembrane proteinsm, cell surface, adhesion-related</span></span>
<span><span class="va">human.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">human.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">human_gene_symbol</span>, <span class="va">human_chromosome</span>, <span class="va">entity_type</span>, <span class="va">category</span>, <span class="va">parent</span>, <span class="va">consensus_score</span>, <span class="va">database_cs</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">human_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">human_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">summarize</span><span class="op">(</span></span>
<span>        human_chromosome <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">human_chromosome</span><span class="op">)</span>,</span>
<span>        entity_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">entity_type</span><span class="op">)</span>,</span>
<span>        category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">category</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        parent <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">parent</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        consensus_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">consensus_score</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>        is_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">parent</span>, <span class="st">"transmembrane|cell_surface|adhesion|junction"</span><span class="op">)</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>        database_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">database_cs</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        <span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize output</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human.omnipath.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># import OmniPathR databases for human transcriptional interactions</span></span>
<span><span class="va">human.regulon.db</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/omnipath-interactions.html" class="external-link">import_transcriptional_interactions</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="fl">9606</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human.regulon.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select columns of interest for gene annotation</span></span>
<span><span class="co"># annotate genes that are transcription factors</span></span>
<span><span class="va">human.regulon.db</span> <span class="op">&lt;-</span> <span class="va">human.regulon.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">source_genesymbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">separate_rows</span><span class="op">(</span><span class="va">source_genesymbol</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">separate_rows</span><span class="op">(</span><span class="va">sources</span>, sep <span class="op">=</span> <span class="st">";"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>human_gene_symbol <span class="op">=</span> <span class="va">source_genesymbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">human_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">summarize</span><span class="op">(</span>database_tf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sources</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">";"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human.regulon.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># merge cell surface and transcription factor annotations</span></span>
<span><span class="va">human.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">human.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>is_tf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">human_gene_symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">human.regulon.db</span><span class="op">$</span><span class="va">human_gene_symbol</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">human.regulon.db</span>, by <span class="op">=</span> <span class="st">"human_gene_symbol"</span>, all.x <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human.omnipath.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># save OmniPathR databases for human gene annotation in TSV file</span></span>
<span><span class="co"># write.table(human.omnipath.db, "inst/extdata/release-105_human_omnipath_db.tsv", col.names = TRUE, row.names = FALSE, quote = FALSE, sep = "\t")</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="build-mouse-omnipath-database">Build mouse omnipath database<a class="anchor" aria-label="anchor" href="#build-mouse-omnipath-database"></a>
</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># import OmniPathR databases for mouse gene annotation</span></span>
<span><span class="va">mouse.omnipath.db</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/intercell.html" class="external-link">import_omnipath_intercell</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="fl">10090</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse.omnipath.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add mouse_gene_symbol and database_cs columns</span></span>
<span><span class="va">mouse.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">mouse.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span></span>
<span>        human_gene_symbol <span class="op">=</span>  <span class="va">genesymbol</span>,</span>
<span>        database_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">database</span>, <span class="st">"UniProt"</span><span class="op">)</span>, <span class="st">"UniProt"</span>, <span class="va">database</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">biomart_dict</span>, <span class="va">.</span>, by <span class="op">=</span> <span class="st">"human_gene_symbol"</span>, all <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"genesymbol"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select columns of interest for gene annotation</span></span>
<span><span class="co"># annotate genes that are transmembrane proteinsm, cell surface, adhesion-related</span></span>
<span><span class="va">mouse.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">mouse.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">mouse_gene_symbol</span>, <span class="va">mouse_chromosome</span>, <span class="va">entity_type</span>, <span class="va">category</span>, <span class="va">parent</span>, <span class="va">consensus_score</span>, <span class="va">database_cs</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">arrange</span><span class="op">(</span><span class="va">mouse_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">mouse_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">summarize</span><span class="op">(</span></span>
<span>        mouse_gene_symbol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mouse_gene_symbol</span><span class="op">)</span>,</span>
<span>        mouse_chromosome <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mouse_chromosome</span><span class="op">)</span>,</span>
<span>        entity_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">entity_type</span><span class="op">)</span>,</span>
<span>        category <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">category</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        parent <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">parent</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        consensus_score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">consensus_score</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>        is_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu">str_detect</span><span class="op">(</span><span class="va">parent</span>, <span class="st">"transmembrane|cell_surface|adhesion|junction"</span><span class="op">)</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>        database_cs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">database_cs</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>,</span>
<span>        <span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">mouse_gene_symbol</span> <span class="op">!=</span> <span class="st">""</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse.omnipath.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># import OmniPathR databases for mouse transcriptional interactions</span></span>
<span><span class="va">mouse.regulon.db</span> <span class="op">&lt;-</span> <span class="fu">OmnipathR</span><span class="fu">::</span><span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/omnipath-interactions.html" class="external-link">import_transcriptional_interactions</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="fl">10090</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse.regulon.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select columns of interest for gene annotation</span></span>
<span><span class="co"># annotate genes that are transcription factors</span></span>
<span><span class="va">mouse.regulon.db</span> <span class="op">&lt;-</span> <span class="va">mouse.regulon.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">source_genesymbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">separate_rows</span><span class="op">(</span><span class="va">source_genesymbol</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">separate_rows</span><span class="op">(</span><span class="va">sources</span>, sep <span class="op">=</span> <span class="st">";"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">mutate</span><span class="op">(</span>mouse_gene_symbol <span class="op">=</span> <span class="va">source_genesymbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">group_by</span><span class="op">(</span><span class="va">mouse_gene_symbol</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">summarize</span><span class="op">(</span>database_tf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">sources</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">";"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse.regulon.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># merge cell surface and transcription factor annotations</span></span>
<span><span class="va">mouse.omnipath.db</span> <span class="op">&lt;-</span> <span class="va">mouse.omnipath.db</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">mouse.regulon.db</span>, by <span class="op">=</span> <span class="st">"mouse_gene_symbol"</span>, all.x <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse.omnipath.db</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># save OmniPathR databases for mouse gene annotation in TSV file</span></span>
<span><span class="co"># write.table(mouse.omnipath.db, "inst/extdata/release-105_mouse_omnipath_db.tsv", col.names = TRUE, row.names = FALSE, quote = FALSE, sep = "\t")</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] strpip_0.1.0     biomaRt_2.62.1   OmnipathR_3.14.0 magrittr_2.0.3  </span></span>
<span><span class="co">## [5] rlang_1.1.5      stringr_1.5.1    tibble_3.2.1     tidyr_1.3.1     </span></span>
<span><span class="co">## [9] dplyr_1.1.4     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] KEGGREST_1.46.0         httr2_1.1.2             xfun_0.51              </span></span>
<span><span class="co">##  [4] bslib_0.9.0             Biobase_2.66.0          tzdb_0.5.0             </span></span>
<span><span class="co">##  [7] vctrs_0.6.5             tools_4.4.3             generics_0.1.3         </span></span>
<span><span class="co">## [10] stats4_4.4.3            curl_6.2.2              AnnotationDbi_1.68.0   </span></span>
<span><span class="co">## [13] RSQLite_2.3.9           blob_1.2.4              pkgconfig_2.0.3        </span></span>
<span><span class="co">## [16] R.oo_1.27.0             checkmate_2.3.2         dbplyr_2.5.0           </span></span>
<span><span class="co">## [19] S4Vectors_0.44.0        desc_1.4.3              readxl_1.4.5           </span></span>
<span><span class="co">## [22] GenomeInfoDbData_1.2.13 lifecycle_1.0.4         compiler_4.4.3         </span></span>
<span><span class="co">## [25] Biostrings_2.74.1       textshaping_1.0.0       progress_1.2.3         </span></span>
<span><span class="co">## [28] GenomeInfoDb_1.42.3     htmltools_0.5.8.1       sass_0.4.9             </span></span>
<span><span class="co">## [31] yaml_2.3.10             pillar_1.10.1           pkgdown_2.1.1          </span></span>
<span><span class="co">## [34] later_1.4.1             crayon_1.5.3            jquerylib_0.1.4        </span></span>
<span><span class="co">## [37] R.utils_2.13.0          cachem_1.1.0            zip_2.3.2              </span></span>
<span><span class="co">## [40] tidyselect_1.2.1        rvest_1.0.4             digest_0.6.37          </span></span>
<span><span class="co">## [43] stringi_1.8.7           purrr_1.0.4             fastmap_1.2.0          </span></span>
<span><span class="co">## [46] cli_3.6.4               logger_0.4.0            XML_3.99-0.18          </span></span>
<span><span class="co">## [49] withr_3.0.2             readr_2.1.5             filelock_1.0.3         </span></span>
<span><span class="co">## [52] UCSC.utils_1.2.0        prettyunits_1.2.0       backports_1.5.0        </span></span>
<span><span class="co">## [55] rappdirs_0.3.3          bit64_4.6.0-1           lubridate_1.9.4        </span></span>
<span><span class="co">## [58] timechange_0.3.0        XVector_0.46.0          rmarkdown_2.29         </span></span>
<span><span class="co">## [61] httr_1.4.7              igraph_2.1.4            bit_4.6.0              </span></span>
<span><span class="co">## [64] cellranger_1.1.0        png_0.1-8               ragg_1.3.3             </span></span>
<span><span class="co">## [67] R.methodsS3_1.8.2       hms_1.1.3               memoise_2.0.1          </span></span>
<span><span class="co">## [70] evaluate_1.0.3          knitr_1.50              IRanges_2.40.1         </span></span>
<span><span class="co">## [73] BiocFileCache_2.14.0    Rcpp_1.0.14             glue_1.8.0             </span></span>
<span><span class="co">## [76] DBI_1.2.3               xml2_1.3.8              BiocGenerics_0.52.0    </span></span>
<span><span class="co">## [79] jsonlite_2.0.0          R6_2.6.1                zlibbioc_1.52.0        </span></span>
<span><span class="co">## [82] systemfonts_1.2.1       fs_1.6.5</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hung M.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
