<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get Started â€¢ strpip</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Started">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">strpip</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/strpip.html">Get started</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/interoperability.html">Interoperability with file formats</a></li>
    <li><a class="dropdown-item" href="../articles/database.html">Build OmniPathR databases</a></li>
    <li><a class="dropdown-item" href="../articles/session_info.html">Session Info</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/hungms/strpip/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get Started</h1>
                        <h4 data-toc-skip class="author">Author: Matthew
Hung</h4>
            
            <h4 data-toc-skip class="date">Last Updated: 2025/04/08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hungms/strpip/blob/main/vignettes/strpip.Rmd" class="external-link"><code>vignettes/strpip.Rmd</code></a></small>
      <div class="d-none name"><code>strpip.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hungms.github.io/deseq2pip/" class="external-link">strpip</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'magrittr'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:rlang':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     set_names</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:tidyr':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     extract</span></span></code></pre>
<pre><code><span><span class="co">## strpip 0.1.1</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load example expression matrix</span></span>
<span><span class="va">data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"GSE189410_expr.txt"</span>, package <span class="op">=</span> <span class="st">"strpip"</span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">data_file</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##               BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1</span></span>
<span><span class="co">## 0610005C13Rik       18        8       18       11       12</span></span>
<span><span class="co">## 0610006L08Rik        0        0       11        0        0</span></span>
<span><span class="co">## 0610009B22Rik      102       93      124      128      126</span></span>
<span><span class="co">## 0610009E02Rik       50       11       22       13       21</span></span>
<span><span class="co">## 0610009L18Rik        5       10        3        4        3</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 53571    11</span></span></code></pre>
<div class="section level2">
<h2 id="remove-xy-mt-vdj-genes">Remove XY, MT, VDJ genes<a class="anchor" aria-label="anchor" href="#remove-xy-mt-vdj-genes"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove xy-related genes from matrix</span></span>
<span><span class="va">xy.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_xy_genes.html">get_xy_genes</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">xy.genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Gm28489" "Gm21867" "Gm29448" "Gm21127" "Gm28488" "Gm20816"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">xy.genes</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 50450    11</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove mt-related genes from matrix</span></span>
<span><span class="va">mt.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mt_genes.html">get_mt_genes</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mt.genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "mt-Tf"   "mt-Rnr1" "mt-Tv"   "mt-Rnr2" "mt-Tl1"  "mt-Nd1"</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mt.genes</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 50426    11</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove vdj-related genes from matrix</span></span>
<span><span class="va">vdj.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_str_genes.html">get_str_genes</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"mouse"</span>, str <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tcr"</span>, <span class="st">"bcr"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">vdj.genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Igll1"    "Trbv13-1" "Trbv12-2" "Trbv13-2" "Trbv12-3" "Trbv13-3"</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">vdj.genes</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 49760    11</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="converting-orthologous-gene-symbols">Converting orthologous gene symbols<a class="anchor" aria-label="anchor" href="#converting-orthologous-gene-symbols"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert mouse gene symbols to human</span></span>
<span><span class="va">human_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_mouse_to_human.html">convert_mouse_to_human</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   mouse_gene_symbol human_gene_symbol</span></span>
<span><span class="co">## 1             Dgcr6                  </span></span>
<span><span class="co">## 2             Prodh                  </span></span>
<span><span class="co">## 3           Gm22809                  </span></span>
<span><span class="co">## 4              Frg1                  </span></span>
<span><span class="co">## 5             Tekt4             MAFIP</span></span>
<span><span class="co">## 6            Mrpl23</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouse_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_human_to_mouse.html">convert_human_to_mouse</a></span><span class="op">(</span><span class="va">human_genes</span><span class="op">$</span><span class="va">human_gene_symbol</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   human_gene_symbol mouse_gene_symbol</span></span>
<span><span class="co">## 1                               Dgcr6</span></span>
<span><span class="co">## 2                               Prodh</span></span>
<span><span class="co">## 3                             Gm22809</span></span>
<span><span class="co">## 4                                Frg1</span></span>
<span><span class="co">## 5             MAFIP             Tekt4</span></span>
<span><span class="co">## 6                              Mrpl23</span></span></code></pre>
<p>Convert human gene symbols to mouse.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert human gene symbols to mouse</span></span>
<span><span class="va">human_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_orthologs_vector.html">convert_orthologs_vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, mode <span class="op">=</span> <span class="st">"mouse_to_human"</span>, one.to.many <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] ""         "MAFIP"    "HSFY2"    "HSFY1"    "RNU2-57P" "DAZ4"</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mouse_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_orthologs_vector.html">convert_orthologs_vector</a></span><span class="op">(</span><span class="va">human_genes</span>, mode <span class="op">=</span> <span class="st">"human_to_mouse"</span>, one.to.many <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mouse_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Dgcr6"   "Prodh"   "Gm22809" "Frg1"    "Tekt4"   "Mrpl23"</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert mouse gene symbols to human</span></span>
<span><span class="va">human_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_orthologs_matrix.html">convert_orthologs_matrix</a></span><span class="op">(</span><span class="va">counts</span>, mode <span class="op">=</span> <span class="st">"mouse_to_human"</span>, many.to.one <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human_counts</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##         BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1</span></span>
<span><span class="co">##            87103    76892    88033    91809    71366</span></span>
<span><span class="co">## A1BG           0        5        1        0        0</span></span>
<span><span class="co">## A1CF           0        0        0        1        0</span></span>
<span><span class="co">## A2M           13       15        1       13        1</span></span>
<span><span class="co">## A3GALT2        3        8        0        0        7</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">human_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_orthologs_matrix.html">convert_orthologs_matrix</a></span><span class="op">(</span><span class="va">counts</span>, mode <span class="op">=</span> <span class="st">"mouse_to_human"</span>, many.to.one <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">human_counts</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##          BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1</span></span>
<span><span class="co">## NCBP2AS2      191      190      197      160      147</span></span>
<span><span class="co">## C2orf68       213      275      339      284      202</span></span>
<span><span class="co">## C4orf19         0        0        0        0        1</span></span>
<span><span class="co">## C4orf54        10        4        9        8        2</span></span>
<span><span class="co">## C11orf58      827      693      825      861      951</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="annotate-gene-function-from-omnipath">Annotate gene function from OmniPath<a class="anchor" aria-label="anchor" href="#annotate-gene-function-from-omnipath"></a>
</h2>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check if OmnipathR is available</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"OmnipathR"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"OmnipathR is not installed. Skipping annotation example."</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="co"># run annotation requires a gene symbol column in the expression matrix</span></span>
<span>  <span class="va">counts</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># run annotation</span></span>
<span>  <span class="va">counts.annotated</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_annotation.html">run_annotation</a></span><span class="op">(</span><span class="va">counts</span>, gene_column <span class="op">=</span> <span class="st">"gene"</span>, org <span class="op">=</span> <span class="st">"mouse"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Check if the annotation was successful</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="st">"entity_type"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts.annotated</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">counts.annotated</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">entity_type</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Annotation data not found or annotation failed. This is expected if OmniPathR is not installed or annotation files are missing."</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">counts.annotated</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">##             gene BM_IgA_1 BM_IgA_2 BM_IgA_3 BM_IgA_4 BM_IgG_1 BM_IgG_2 BM_IgG_3</span></span>
<span><span class="co">## 6  0610010K14Rik       76       86       88      119       64       29       55</span></span>
<span><span class="co">## 10 0610030E20Rik      213      275      339      284      202      214      246</span></span>
<span><span class="co">## 17 0610040J01Rik        0        0        0        0        1        0        1</span></span>
<span><span class="co">## 41 1110032A03Rik      223      241      271      211      194      246      229</span></span>
<span><span class="co">## 42 1110032F04Rik        3       20       16        2        0        0        3</span></span>
<span><span class="co">## 45 1110038F14Rik      258      224      247      285      186      207      220</span></span>
<span><span class="co">##    BM_IgG_4 BM_IgM_1 BM_IgM_3 BM_IgM_4 mouse_chromosome human_gene_symbol</span></span>
<span><span class="co">## 6        86      106      118      157               11   RNASEK-C17orf49</span></span>
<span><span class="co">## 10      306      267      519      422                6           C2orf68</span></span>
<span><span class="co">## 17        0        0        0        3                5           C4orf19</span></span>
<span><span class="co">## 41      190      454      439      243                9           C11orf1</span></span>
<span><span class="co">## 42        0        0        1        1                3           C3orf80</span></span>
<span><span class="co">## 45      217      282      341      283               15           C8orf33</span></span>
<span><span class="co">##    human_chromosome human_gene_symbol_unique human_chromosome_unique</span></span>
<span><span class="co">## 6                17          RNASEK-C17orf49                      17</span></span>
<span><span class="co">## 10                2                  C2orf68                       2</span></span>
<span><span class="co">## 17                4                  C4orf19                       4</span></span>
<span><span class="co">## 41               11                  C11orf1                      11</span></span>
<span><span class="co">## 42                3                  C3orf80                       3</span></span>
<span><span class="co">## 45                8                  C8orf33                       8</span></span>
<span><span class="co">##    entity_type                    category</span></span>
<span><span class="co">## 6      protein               intracellular</span></span>
<span><span class="co">## 10     protein               intracellular</span></span>
<span><span class="co">## 17     protein               intracellular</span></span>
<span><span class="co">## 41     protein               intracellular</span></span>
<span><span class="co">## 42     protein transmembrane, cell_surface</span></span>
<span><span class="co">## 45     protein               intracellular</span></span>
<span><span class="co">##                                                  parent consensus_score is_cs</span></span>
<span><span class="co">## 6                                         intracellular               1 FALSE</span></span>
<span><span class="co">## 10                                        intracellular               1 FALSE</span></span>
<span><span class="co">## 17                                        intracellular               1 FALSE</span></span>
<span><span class="co">## 41                                        intracellular               3 FALSE</span></span>
<span><span class="co">## 42 transmembrane_predicted, cell_surface, transmembrane               4  TRUE</span></span>
<span><span class="co">## 45                                        intracellular               1 FALSE</span></span>
<span><span class="co">##                                                    database_cs is_tf</span></span>
<span><span class="co">## 6                                             OmniPath, ComPPI FALSE</span></span>
<span><span class="co">## 10                                            OmniPath, ComPPI FALSE</span></span>
<span><span class="co">## 17                                            ComPPI, OmniPath FALSE</span></span>
<span><span class="co">## 41                     ComPPI, OmniPath, UniProt, GO_Intercell FALSE</span></span>
<span><span class="co">## 42 Phobius, OmniPath, Ramilowski_location, Surfaceome, UniProt FALSE</span></span>
<span><span class="co">## 45                                            ComPPI, OmniPath FALSE</span></span>
<span><span class="co">##    database_tf</span></span>
<span><span class="co">## 6         &lt;NA&gt;</span></span>
<span><span class="co">## 10        &lt;NA&gt;</span></span>
<span><span class="co">## 17        &lt;NA&gt;</span></span>
<span><span class="co">## 41        &lt;NA&gt;</span></span>
<span><span class="co">## 42        &lt;NA&gt;</span></span>
<span><span class="co">## 45        &lt;NA&gt;</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.3 (2025-02-28)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] strpip_0.1.1     biomaRt_2.62.1   OmnipathR_3.14.0 magrittr_2.0.3  </span></span>
<span><span class="co">## [5] rlang_1.1.5      stringr_1.5.1    tibble_3.2.1     tidyr_1.3.1     </span></span>
<span><span class="co">## [9] dplyr_1.1.4     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] KEGGREST_1.46.0         httr2_1.1.2             xfun_0.52              </span></span>
<span><span class="co">##  [4] bslib_0.9.0             Biobase_2.66.0          tzdb_0.5.0             </span></span>
<span><span class="co">##  [7] vctrs_0.6.5             tools_4.4.3             generics_0.1.3         </span></span>
<span><span class="co">## [10] stats4_4.4.3            curl_6.2.2              AnnotationDbi_1.68.0   </span></span>
<span><span class="co">## [13] RSQLite_2.3.9           blob_1.2.4              pkgconfig_2.0.3        </span></span>
<span><span class="co">## [16] R.oo_1.27.0             checkmate_2.3.2         dbplyr_2.5.0           </span></span>
<span><span class="co">## [19] S4Vectors_0.44.0        desc_1.4.3              readxl_1.4.5           </span></span>
<span><span class="co">## [22] GenomeInfoDbData_1.2.13 lifecycle_1.0.4         compiler_4.4.3         </span></span>
<span><span class="co">## [25] Biostrings_2.74.1       textshaping_1.0.0       progress_1.2.3         </span></span>
<span><span class="co">## [28] GenomeInfoDb_1.42.3     htmltools_0.5.8.1       sass_0.4.9             </span></span>
<span><span class="co">## [31] yaml_2.3.10             pillar_1.10.2           pkgdown_2.1.1          </span></span>
<span><span class="co">## [34] later_1.4.1             crayon_1.5.3            jquerylib_0.1.4        </span></span>
<span><span class="co">## [37] R.utils_2.13.0          cachem_1.1.0            zip_2.3.2              </span></span>
<span><span class="co">## [40] tidyselect_1.2.1        rvest_1.0.4             digest_0.6.37          </span></span>
<span><span class="co">## [43] stringi_1.8.7           purrr_1.0.4             fastmap_1.2.0          </span></span>
<span><span class="co">## [46] cli_3.6.4               logger_0.4.0            XML_3.99-0.18          </span></span>
<span><span class="co">## [49] withr_3.0.2             readr_2.1.5             filelock_1.0.3         </span></span>
<span><span class="co">## [52] UCSC.utils_1.2.0        prettyunits_1.2.0       backports_1.5.0        </span></span>
<span><span class="co">## [55] rappdirs_0.3.3          bit64_4.6.0-1           lubridate_1.9.4        </span></span>
<span><span class="co">## [58] timechange_0.3.0        XVector_0.46.0          rmarkdown_2.29         </span></span>
<span><span class="co">## [61] httr_1.4.7              igraph_2.1.4            bit_4.6.0              </span></span>
<span><span class="co">## [64] cellranger_1.1.0        png_0.1-8               ragg_1.3.3             </span></span>
<span><span class="co">## [67] R.methodsS3_1.8.2       hms_1.1.3               memoise_2.0.1          </span></span>
<span><span class="co">## [70] evaluate_1.0.3          knitr_1.50              IRanges_2.40.1         </span></span>
<span><span class="co">## [73] BiocFileCache_2.14.0    Rcpp_1.0.14             glue_1.8.0             </span></span>
<span><span class="co">## [76] DBI_1.2.3               xml2_1.3.8              BiocGenerics_0.52.0    </span></span>
<span><span class="co">## [79] jsonlite_2.0.0          R6_2.6.1                zlibbioc_1.52.0        </span></span>
<span><span class="co">## [82] systemfonts_1.2.2       fs_1.6.5</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hung M.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
